src/cs50/sql.py

```diff
@@ -29,25 +56,23 @@ def __init__(self, url, **kwargs):
             if not os.path.isfile(matches.group(1)):
                 raise RuntimeError("not a file: {}".format(matches.group(1)))
 
-            # Remember foreign_keys and remove it from kwargs
-            foreign_keys = kwargs.pop("foreign_keys", False)
+        # Create engine, disabling SQLAlchemy's own autocommit mode, raising exception if back end's module not installed
+        self._engine = sqlalchemy.create_engine(url, **kwargs).execution_options(autocommit=False)
 
-            # Create engine, raising exception if back end's module not installed
-            self.engine = sqlalchemy.create_engine(url, **kwargs)
+        # Listener for connections
+        def connect(dbapi_connection, connection_record):
 
-            # Enable foreign key constraints
-            if foreign_keys:
-                def connect(dbapi_connection, connection_record):
-                    if type(dbapi_connection) is sqlite3.Connection:  # If back end is sqlite
-                        cursor = dbapi_connection.cursor()
-                        cursor.execute("PRAGMA foreign_keys=ON")
-                        cursor.close()
-                sqlalchemy.event.listen(self.engine, "connect", connect)
+            # Disable underlying API's own emitting of BEGIN and COMMIT
+            dbapi_connection.isolation_level = None
```

Comment:
Seems to be the empty string. Without an explicit `None`, you get

```
cannot start a transaction within a transaction
```

for code like:

```
    def test_rollback(self):
        self.db.execute("BEGIN")
        self.db.execute("INSERT INTO cs50 (val) VALUES('foo')")
        self.db.execute("INSERT INTO cs50 (val) VALUES('bar')")
        self.db.execute("ROLLBACK")
        self.assertEqual(self.db.execute("SELECT val FROM cs50"), [])
```

And creating the engine with 

```
sqlalchemy.create_engine(url, **kwargs).execution_options(autocommit=False, isolation_level=None)
```

yields

```
AttributeError: 'NoneType' object has no attribute 'replace'
```

since SQLAlchemy's API seems to expect only a string. Seems to be the underlying API(s) that accept `None`.